internal/runners/packages/search.go

```diff
@@ -71,7 +83,46 @@ func (s *Search) Run(params SearchRunParams, nstype model.NamespaceType) error {
 		)
 	}
 
-	s.out.Print(output.Prepare(formatSearchResults(packages, params.Ingredient.Namespace != ""), packages))
+	// The search endpoint will return all versions of a package, so we need to
+	// use only the latest version of each package.
+	seen := make(map[string]bool)
+	var processedPackages []*model.IngredientAndVersion
+	for _, pack := range packages {
+		if pack.Ingredient.Name == nil {
+			logging.Error("Package has no name: %v", pack)
+			continue
+		}
+
+		if seen[*pack.Ingredient.Name] {
+			continue
+		}
+		processedPackages = append(processedPackages, pack)
+		seen[*pack.Ingredient.Name] = true
+	}
+
+	var vulns map[string][]*model.VulnerabilityIngredient
+	if s.auth.Authenticated() {
+		vulns, err = s.getVulns(processedPackages)
+		if err != nil {
+			return errs.Wrap(err, "Could not fetch vulnerabilities")
+		}
+	}
+
+	table, err := createSearchTable(v.width, v.height, processedPackages, vulns)
+	if err != nil {
+		return errs.Wrap(err, "Could not create search table")
+	}
+	v.content = table.Content()
+	v.packageNames = table.packageNames
```

Comment:
I've updated this a bit so that the view is more encapsulated. We are still using this type to process the search results but any view processing for the viewport is now done elsewhere.